src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -431,16 +445,55 @@ internal void ParseSolution()
 
             ParseFileHeader();
 
+            HashSet<string> slnFilter = null;
+            if (_solutionFilterFile != null)
+            {
+                slnFilter = new HashSet<string>();
+                string filterFile = File.ReadAllText(_solutionFilterFile);
+                int projectsInd = filterFile.IndexOf('[', filterFile.IndexOf("\"projects\":"));
+                bool inQuote = false;
+                StringBuilder sb = null;
+                if (projectsInd == -1)
+                {
+                    throw new InvalidDataException("Solution filter file is incorrectly formatted.");
```

Comment:
These need to produce localized user-facing errors with MSB codes. How are other sln-parsing errors handled? Can we do things the same way?